epochStart/metachain/stakingDataProvider.go

```diff
@@ -239,32 +269,123 @@ func (sdp *stakingDataProvider) getTopUpValueAndTotalStaked(validatorAddress str
 			GasProvided: math.MaxUint64,
 		},
 		RecipientAddr: vm.ValidatorSCAddress,
-		Function:      "getTopUpTotalStaked",
+		Function:      "getTotalStakedTopUpBlsKeys",
 	}
 
 	vmOutput, err := sdp.systemVM.RunSmartContractCall(vmInput)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	if vmOutput.ReturnCode != vmcommon.Ok {
-		return nil, nil, fmt.Errorf("%w, error: %v", epochStart.ErrExecutingSystemScCode, vmOutput.ReturnCode)
+		return nil, nil, nil, fmt.Errorf("%w, error: %v", epochStart.ErrExecutingSystemScCode, vmOutput.ReturnCode)
 	}
-	topUpBytes := vmOutput.ReturnData
-	if len(topUpBytes) != 2 {
-		return nil, nil, fmt.Errorf("%w, getTopUp function should have returned exactly two values: the top up value and total staked values", epochStart.ErrExecutingSystemScCode)
+
+	if len(vmOutput.ReturnData) < 2 {
+		return nil, nil, nil, fmt.Errorf("%w, getTotalStakedTopUpBlsKeys function should have at least two values", epochStart.ErrExecutingSystemScCode)
 	}
 
-	topUpValue, ok := big.NewInt(0).SetString(string(topUpBytes[0]), conversionBase)
+	topUpValue, ok := big.NewInt(0).SetString(string(vmOutput.ReturnData[0]), conversionBase)
 	if !ok {
-		return nil, nil, fmt.Errorf("%w, error: topUp string returned is not a number", epochStart.ErrExecutingSystemScCode)
+		return nil, nil, nil, fmt.Errorf("%w, error: topUp string returned is not a number", epochStart.ErrExecutingSystemScCode)
 	}
 
-	totalStakedValue, ok := big.NewInt(0).SetString(string(topUpBytes[1]), conversionBase)
+	totalStakedValue, ok := big.NewInt(0).SetString(string(vmOutput.ReturnData[1]), conversionBase)
 	if !ok {
-		return nil, nil, fmt.Errorf("%w, error: totalStaked string returned is not a number", epochStart.ErrExecutingSystemScCode)
+		return nil, nil, nil, fmt.Errorf("%w, error: totalStaked string returned is not a number", epochStart.ErrExecutingSystemScCode)
+	}
+
+	return topUpValue, totalStakedValue, vmOutput.ReturnData[2:], nil
```

Comment:
Are the "leaving" keys also among the returned keys here?

Reply:
All the blsKeys are returned. Everything which was registered